### PR TITLE
run ev docker directives as root

### PR DESCRIPTION
**Pull request by doylemark**
# Why
When "USER" directives are present in the user's Dockerfile, ev injected directives could have insufficient permissions

# How
Run ev injected directives as root (default for dockerfile), then insert `su` command into user-service/run

### CODE REVIEW

File: src/build/mod.rs [hunk 1]
```diff
@@ -165,6 +168,11 @@ async fn process_dockerfile<R: AsyncRead + std::marker::Unpin>(
             last_entrypoint = Some(directive.clone());
         } else if let Directive::Expose { port } = directive {
             exposed_port = *port;
+        } else if let Directive::User(b) = directive{
+            if let Ok(user) = String::from_utf8(b.to_vec()) {
```
Comment (doylemark) [inline src/build/mod.rs:172]: should we maybe return a build error here if we can't parse the user directive? (not sure)

File: src/build/mod.rs [hunk 2]
```diff
@@ -307,6 +306,34 @@ fn reproducible_build_directives() -> Vec<Directive> {
...
 }
 
+pub fn build_user_service(entrypoint: String, wait_for_env: &str, last_user: String) -> Directive {
+    let su_cmd = format!("su {last_user}");
+    let exec_cmd = format!("exec {}", entrypoint);
+
+    let cmds = vec![
+        su_cmd.as_str(),
+        "sleep 5",
+        "echo \\\"Checking status of data-plane\\\"",
```
Comment (doylemark) [inline src/build/mod.rs]: interested to know why these need to be triple escaped? (just copied from og implementation)

File: src/build/mod.rs [hunk 3]
```diff
@@ -156,6 +156,7 @@ async fn process_dockerfile<R: AsyncRead + std::marker::Unpin>(
     // Filter out unwanted directives
     let mut last_cmd = None;
     let mut last_entrypoint = None;
+    let mut last_user = "root".to_string();
```
Comment (lfarrel6) [inline src/build/mod.rs]: `last_user` should start as `None` to avoid unneeded switching

File: src/build/mod.rs (hunk 1 above)
Comment (lfarrel6) [reply src/build/mod.rs:172]: yeah throwing an error here makes sense

File: src/build/mod.rs (hunk 2 above)
Comment (lfarrel6) [reply src/build/mod.rs]: it's just escaping the two characters (`/` -> `\\` and `"` -> `\"`). The strings here are then printed to a file in the Cage, which results in `echo "..."` being written.

Would be cleaner to use raw strings tho:
```suggestion
        r#"echo \"Checking status of data-plane\""#,
```

**Review by lfarrel6 — COMMENTED** [review]
looks good, couple suggestions

**Review by lfarrel6 — APPROVED** [review]

**Pull request merged**